service/rpc/header.go

```diff
@@ -11,13 +11,32 @@ import (
 )
 
 const (
+	headEndpoint           = "/head"
 	headerByHeightEndpoint = "/header"
 )
 
 var (
 	heightKey = "height"
 )
 
+func (h *Handler) handleHeadRequest(w http.ResponseWriter, r *http.Request) {
+	head, err := h.header.Head(r.Context())
```

Comment:
Pls double check that this Context is getting canceled once the processing of the request ends with error or success or anything else. If is not closed automatically, then we need to add to each handler:
```go
ctx, cancel := context.WithCancel(r.Context())
defer cancel()
```

Reply:
https://github.com/celestiaorg/celestia-node/issues/726

for a later PR.